pkg/apis/components/v1alpha1/types.go

```diff
@@ -41,6 +41,16 @@ func (Component) Kind() string {
 	return "Component"
 }
 
+// Name returns the component name.
+func (c Component) GetName() string {
+	return c.Name
+}
+
+// Namespace returns the component namespace.
+func (c Component) GetNamespace() string {
+	return c.Namespace
+}
+
 // LogName returns the name of the component that can be used in logging.
 func (c Component) LogName() string {
```

Comment:
I think this should not be changed in this PR as it would cause a lot of unrelated changes. Although if we do want to add a Stringer method *as well* it may not be a bad suggestion, but not high priority. Like:

```go
func (c Component) String() string {
	return c.LogName()
}
```

Reply:
I didn't introduce the `LogName` func in this PR so maybe this should be moved to a separate PR.